src/solvers/solver_utils.hpp

```diff
@@ -135,6 +145,155 @@ struct Stencil {
+      pack.GetNBlocks() - 1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        const int nvars = pack.GetUpperBound(b, in()) - pack.GetLowerBound(b, in()) + 1;
+        for (int c = 0; c < nvars; ++c)
+          pack(b, te, out(c), k, j, i) = pack(b, te, in(c), k, j, i);
+      });
+  return TaskStatus::complete;
+}
+
+template <class a_t, class b_t, class out, bool only_fine_on_composite = true>
+TaskStatus AddFieldsAndStoreInteriorSelect(const std::shared_ptr<MeshData<Real>> &md,
+                                           Real wa = 1.0, Real wb = 1.0,
+                                           bool only_interior = false) {
+  using TE = parthenon::TopologicalElement;
+  TE te = TE::CC;
+  IndexRange ib = md->GetBoundsI(IndexDomain::entire, te);
+  IndexRange jb = md->GetBoundsJ(IndexDomain::entire, te);
+  IndexRange kb = md->GetBoundsK(IndexDomain::entire, te);
+
+  int nblocks = md->NumBlocks();
+  std::vector<bool> include_block(nblocks, true);
+  if (only_interior) {
+    // The neighbors array will only be set for a block if its a leaf block
+    for (int b = 0; b < nblocks; ++b)
+      include_block[b] = md->GetBlockData(b)->GetBlockPointer()->neighbors.size() == 0;
+  }
+
+  auto desc = parthenon::MakePackDescriptor<a_t, b_t, out>(md.get());
+  auto pack = desc.GetPack(md.get(), include_block, only_fine_on_composite);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "SetPotentialToZero", DevExecSpace(), 0,
+      pack.GetNBlocks() - 1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        const int nvars = pack.GetUpperBound(b, a_t()) - pack.GetLowerBound(b, a_t()) + 1;
+        for (int c = 0; c < nvars; ++c) {
+          pack(b, te, out(c), k, j, i) =
+              wa * pack(b, te, a_t(c), k, j, i) + wb * pack(b, te, b_t(c), k, j, i);
+        }
+      });
+  return TaskStatus::complete;
+}
+
+template <class a_t, class b_t, class out, bool only_fine_on_composite = true>
+TaskStatus AddFieldsAndStore(const std::shared_ptr<MeshData<Real>> &md, Real wa = 1.0,
+                             Real wb = 1.0) {
+  return AddFieldsAndStoreInteriorSelect<a_t, b_t, out, only_fine_on_composite>(
+      md, wa, wb, false);
+}
+
+template <class var, bool only_fine_on_composite = true>
+TaskStatus SetToZero(const std::shared_ptr<MeshData<Real>> &md) {
+  int nblocks = md->NumBlocks();
+  using TE = parthenon::TopologicalElement;
+  TE te = TE::CC;
+  std::vector<bool> include_block(nblocks, true);
+  auto desc = parthenon::MakePackDescriptor<var>(md.get());
+  auto pack = desc.GetPack(md.get(), include_block, only_fine_on_composite);
+  const size_t scratch_size_in_bytes = 0;
+  const int scratch_level = 1;
+  const int ng = parthenon::Globals::nghost;
+  parthenon::par_for_outer(
+      DEFAULT_OUTER_LOOP_PATTERN, "Print", DevExecSpace(), scratch_size_in_bytes,
```

Comment:
Ironically also incorrect kernel name... since all the others are call` SetToZero`

Reply:
haha, not sure how that happened. fixed.